Sources/Extensions/UIKit/UIViewExtensions.swift

```diff
@@ -224,7 +215,25 @@ public extension UIView {
 // MARK: - Methods
 public extension UIView {
 
-	/// SwifterSwift: Set some or all corners radiuses of view.
+    /// SwifterSwift: Recursively find the first responder.
+    public func firstResponder() -> UIView? {
+        guard
```

Comment:
Wouldn't this work?
```
public var firstResponder: UIView? {
    guard !isFirstResponder else { return self }

    return subviews.first(where: { $0.firstResponder != nil })
}
```
It would do a recursive, depth-first search for the first responder.

Reply:
Also, doing it this way you wouldn't need to turn it into a function and we can avoid a breaking change.

Reply:
@guykogus Thank's for the feedback :))
But actually, the way is implemented is because it's better for performance, in the way you mentioned if you have a first responder as the last subview, for example, it will navigate to the whole view hierarchy tree for the other subviews before gets there

Reply:
So my way is a depth-first search. Logically it's the exact same as what you've done.
Alternately we could do it as a breadth-first search, but that's a bit uglier.
```
public var firstResponder: UIView? {
    var firstResponder: UIView?

    var views = [self]
    var i = 0
    while i < views.count, firstResponder == nil {
        let view = views[i]
        if view.isFirstResponder {
            firstResponder = view
        } else {
            views.append(contentsOf: view.subviews)
            i += 1
        }
    }
    return firstResponder
}
```
The question is, why would one method be better than the other? Is there any reason why search BF would be faster than DF?

Reply:
But it's still not a breadth-first search, it's a BFS/DFS hybrid. It's a poorly written algorithm. It double-checks all the subviews, which isn't the right way to do it.